src/Server/HTTPHandler.cpp

```diff
@@ -923,10 +928,18 @@ HTTPRequestHandlerFactoryPtr createDynamicHandlerFactory(IServer & server,
 {
     auto query_param_name = config.getString(config_prefix + ".handler.query_param_name", "query");
 
+    std::optional<BasicCredentials> config_credentials;
+    if (config.has(config_prefix + ".handler.user") || config.has(config_prefix + ".handler.password"))
+    {
+        config_credentials.emplace(
+            config.getString(config_prefix + ".handler.user", "default"),
+            config.getString(config_prefix + ".handler.password", ""));
```

Comment:
I have an alternative idea - instead of building credentials here and passing them to `authenticateUserByHTTP()`, you could pass `config_prefix + ".handler"` to `authenticateUserByHTTP()`  and let `authenticateUserByHTTP()` to read those values from the config by itself.